### DISCUSSION TITLE
Considering Observable Framework, but before I dive in: can selections presented to a user update, based on other selections they make, even in a statically generated OF app?

### DISCUSSION THREAD

**Discussion by bzm3r** [observablehq/framework#1835] [Q&A] [answered]
Here's an issue I run into with other statically generated dashboards: suppose that I have data that has two columns that can be used to categorize it: `channel` (e.g. "A", "B", "C") and `flavor` ("spicy", "mild", "sweet"). 

We can give the user a dropdown for `channel` and `flavor`, letting them choose amongst the available choices. We filter the data itself first by `channel`, and *then* by `flavor`, because not all `flavor`s are available for each `channel`.

I want the selections available for `flavor` to update based on the choice made by the user for `channel`. With something simple like Quarto + Altair, this is not possible. 

Is it possible with Observable Framework?

**Comment by sbarre** [comment, marked as answer]
If you mean can you update data based on user selections in page, then yes, a Framework application is reactive:

https://observablehq.com/framework/reactivity

There are built-in tools and inputs for giving user controls that will cause graphs and page elements to re-render based on those choices.

The specific behaviour you are describing (faceted filtering) is also possible to implement, but there's no built-in tooling for that. 

So nothing stopping you from doing it, but you'd be doing it.
